File: java/src/main/java/chapter12/stubbing/ConcreteCollaborator.java
package chapter12.stubbing;

import java.time.LocalTime;

/**
 * Concrete implementation of the supposed collaborator with a method
 * that returns something unpredictable.
 */
public class ConcreteCollaborator implements Collaborator {

    @Override
    public int computeAndReturnValue() {
        return LocalTime.now().getHour() * 60;
    }
}

File: csharp/DeveloperTestingSamples/Samples/Chapter12/Stubbing/ConcreteCollaborator.cs
using System;

namespace Samples.Chapter12.Stubbing
{

    
    // Concrete implementation of the supposed collaborator with a method
    // that returns something unpredictable.
    public class ConcreteCollaborator : ICollaborator
    {
        public int ComputeAndReturnValue()
        {
            return DateTime.Now.Hour * 60;
        }
    }
}
